apps/merkle_patricia_tree/lib/merkle_patricia_tree/trie/helper.ex

```diff
@@ -40,4 +40,9 @@ defmodule MerklePatriciaTree.Trie.Helper do
   def get_binary(l) do
     for x <- l, into: <<>>, do: <<x::4>>
   end
+
+  # Encodes `x` in RLP if it isn't already encoded.
+  # And it is definitely not encoded if it is `<<>>` or not a binary (e.g. array).
+  def rlp_encode(x) when not is_binary(x) or x == <<>>, do: ExRLP.encode(x)
```

Comment:
`<<>>` is a binary, right? So is the `or x == <<>>` superfluous here?

Reply:
Actually, `is_binary` does not match `<<>>`

Reply:
That's weird. I just tried it on `iex`, 

```elixir
iex> is_binary(<<>>)
true
```

Reply:
you're right